gradle/lib/dependabot/gradle/version.rb

```diff
@@ -28,7 +28,7 @@ class Version < Gem::Version
       VERSION_PATTERN =
         "[0-9a-zA-Z]+"\
         '(?>\.[0-9a-zA-Z]*)*'\
-        '([_-][0-9A-Za-z_-]*(\.[0-9A-Za-z_-]*)*)?'
+        '([_\-\+][0-9A-Za-z_-]*(\.[0-9A-Za-z_-]*)*)?'
```

Comment:
This looks like an omission, brings this pattern to par with `maven`. This is currently tested in the requirement class but without this change it's not `correct?` so you can't initialise it as a version: https://github.com/dependabot/dependabot-core/blob/1ba226ad7fb22f22fa3eef73843e6f6c956573f5/gradle/spec/dependabot/gradle/requirement_spec.rb#L55